patacrep/content/__init__.py

```diff
@@ -241,25 +261,25 @@ def process_content(content, config=None):
     """
     contentlist = ContentList()
     plugins = config.get('_content_plugins', {})
-    keyword_re = re.compile(r'^ *(?P<keyword>[\w\*]*) *(\((?P<argument>.*)\))? *$')
     if not content:
-        content = [["song"]]
+        content = [{'song': None}]
+    elif isinstance(content, dict):
+        content = [content]
     for elem in content:
         if isinstance(elem, str):
-            elem = ["song", elem]
-        try:
-            match = keyword_re.match(elem[0]).groupdict()
-        except AttributeError:
-            contentlist.append_error(ContentError(elem[0], "Cannot parse content type."))
-            continue
-        (keyword, argument) = (match['keyword'], match['argument'])
-        if keyword not in plugins:
-            contentlist.append_error(ContentError(keyword, "Unknown content type."))
-            continue
-        contentlist.extend(plugins[keyword](
-            keyword,
-            argument=argument,
-            contentlist=elem[1:],
-            config=config,
-            ))
+            elem = {'song': elem}
+        if isinstance(elem, dict):
+            for keyword, argument in elem.items():
+                try:
+                    if keyword not in plugins:
+                        raise ContentError(keyword, "Unknown content keyword.")
+                    contentlist.extend(plugins[keyword](
+                        keyword,
+                        argument=argument,
+                        config=config,
+                        ))
+                except ContentError as error:
+                    contentlist.append_error(error)
```

Comment:
Je trouve ça bizarre, cette exception qu'on rattrape immédiatement. J'aurais plutôt fait :

~~~python
for keyword, argument in elem.items():
    if keyword not in plugins:
        contentlist.append_error(ContentError(keyword, "Unknown content keyword"))
        continue
    contentlist.extend(...)